src/libaktualizr/primary/aktualizr.cc

```diff
@@ -170,13 +170,30 @@ boost::signals2::connection Aktualizr::SetSignalHandler(
   return sig_->connect(handler);
 }
 
-std::unique_ptr<StorageTargetRHandle> Aktualizr::GetStoredTarget(const Uptane::Target &target) {
+std::vector<Uptane::Target> Aktualizr::GetStoredTargets() {
+  try {
+    return storage_->getTargetFiles();
+  } catch (...) {
+    return {};
+  }
+}
+
+bool Aktualizr::DeleteStoredTarget(const Uptane::Target &target) {
+  try {
+    storage_->removeTargetFile(target.filename());
+  } catch (...) {
+    return false;
+  }
+  return true;
+}
+
+std::unique_ptr<StorageTargetRHandle> Aktualizr::OpenStoredTarget(const Uptane::Target &target) {
   try {
     auto handle = storage_->openTargetFile(target);
     if (handle->isPartial()) {
       return std::unique_ptr<StorageTargetRHandle>();
     }
-    return storage_->openTargetFile(target);
+    return handle;
```

Comment:
Maybe it would be better to throw an exception if somebody tries to open a not or partially downloaded target?